File: frontend/src/api/notification.ts
/**
 * 通知API
 */
import { get, put, del } from './request'

/**
 * 通知接口
 */
export interface Notification {
  id: number
  userId: number
  type: 'audit_pass' | 'audit_reject' | 'comment' | 'like' | 'system'
  title: string
  content: string
  cardId?: number
  cardTitle?: string
  isRead: number  // 0-未读，1-已读
  createTime: string
  readTime?: string
}

/**
 * 获取用户通知列表
 */
export const getNotifications = (userId: number, limit?: number) => {
  const params: any = { userId }
  if (limit) params.limit = limit
  return get<Notification[]>('/notification/list', params)
}

/**
 * 获取未读通知数量
 */
export const getUnreadCount = (userId: number) => {
  return get<{ count: number }>('/notification/unread-count', { userId })
}

/**
 * 标记单个通知为已读
 */
export const markNotificationAsRead = (notificationId: number, userId: number) => {
  return put<boolean>(`/notification/read/${notificationId}?userId=${userId}`)
}

/**
 * 标记所有通知为已读
 */
export const markAllNotificationsAsRead = (userId: number) => {
  return put<boolean>(`/notification/read-all?userId=${userId}`)
}

/**
 * 删除通知
 */
export const deleteNotification = (notificationId: number, userId: number) => {
  return del<boolean>(`/notification/delete/${notificationId}?userId=${userId}`)
}

File: backend/src/main/java/com/horizon/entity/Notification.java
package com.horizon.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 通知实体
 */
@Data
@TableName("t_notification")
public class Notification {
    
    /**
     * 通知ID
     */
    @TableId(type = IdType.AUTO)
    private Long id;
    
    /**
     * 接收用户ID
     */
    private Long userId;
    
    /**
     * 通知类型：audit_pass(审核通过)、audit_reject(审核拒绝)、comment(评论)、like(点赞)、system(系统通知)
     */
    private String type;
    
    /**
     * 通知标题
     */
    private String title;
    
    /**
     * 通知内容
     */
    private String content;
    
    /**
     * 关联卡片ID（可选）
     */
    private Long cardId;
    
    /**
     * 关联卡片标题（冗余存储，方便显示）
     */
    private String cardTitle;
    
    /**
     * 是否已读：0-未读，1-已读
     */
    private Integer isRead;
    
    /**
     * 创建时间
     */
    private LocalDateTime createTime;
    
    /**
     * 阅读时间
     */
    private LocalDateTime readTime;
}
